src/v2i-hub/CARMAStreetsPlugin/src/J3224ToSDSMJsonConverter.cpp

```diff
@@ -0,0 +1,179 @@
+                            {
+                                case PropelledInformation_PR_NOTHING:
+                                    break;
+                                case PropelledInformation_PR_human:
+                                    optionalDataJson["detected_vru_data"]["propulsion"]["human"] = det_object->detObjOptData->choice.detVRU.propulsion->choice.human;
+                                    break;
+                                case PropelledInformation_PR_animal:
+                                    optionalDataJson["detected_vru_data"]["propulsion"]["human"] = det_object->detObjOptData->choice.detVRU.propulsion->choice.animal;
+                                    break;
+                                case PropelledInformation_PR_motor:
+                                    optionalDataJson["detected_vru_data"]["propulsion"]["human"] = det_object->detObjOptData->choice.detVRU.propulsion->choice.motor;
+                                    break;
+                            }
+
+                            optionalDataJson["detected_vru_data"]["attachment"] = *det_object->detObjOptData->choice.detVRU.attachment;
+                            optionalDataJson["detected_vru_data"]["radius"] = *det_object->detObjOptData->choice.detVRU.radius;
+                            break;
+
+                        case DetectedObjectOptionalData_PR_detObst:
+                            optionalDataJson["detected_obstacle_data"]["obst_size"]["width"] = det_object->detObjOptData->choice.detObst.obstSize.width;
+                            optionalDataJson["detected_obstacle_data"]["obst_size"]["length"] = det_object->detObjOptData->choice.detObst.obstSize.length;
+                            optionalDataJson["detected_obstacle_data"]["obst_size"]["height"] = *det_object->detObjOptData->choice.detObst.obstSize.height; // optional
+
+                            optionalDataJson["detected_obstacle_data"]["obst_size_confidence"]["width_confidence"] = det_object->detObjOptData->choice.detObst.obstSizeConfidence.widthConfidence;
+                            optionalDataJson["detected_obstacle_data"]["obst_size_confidence"]["length_confidence"] = det_object->detObjOptData->choice.detObst.obstSizeConfidence.lengthConfidence;
+                            optionalDataJson["detected_obstacle_data"]["obst_size_confidence"]["height_confidence"] = *det_object->detObjOptData->choice.detObst.obstSizeConfidence.heightConfidence;
+                            break;
+                    }
+                    detectedObjectJson["detected_object_data"]["detected_object_optional_data"] = optionalDataJson;
+                }
+
+                // Add the generated object jsons to the detected object list
+                SDSMDataJson["objects"].append(detectedObjectJson);
+
+            }
+        }
+
+        //std::cout << SDSMDataJson.toStyledString() << std::endl;
```

Comment:
Does this method still exist. Any reason why we would keep this commented out line in the final commit?